### DISCUSSION TITLE
socket.on('disconnect') run with order

### DISCUSSION THREAD

**Discussion by sajjad-fatehi** [socketio/socket.io#4093] [Q&A] [answered]
hi everyone .
imagine i store my peers in redis for each of my rooms .
user1-> redis.set(roomId,[...redis.get(roomId),user1]
and on disconnect:
 redis.set(roomId,[redis.get(roomId).filter(q=>q !== user1)]

for example my 20 users disconnect in same time ... 20 times this code will run in same time and my array will have wrong data

what can i do if i want disconnect event runs one by one ... not in same time

thnks

**Comment by darrachequesne** [comment, marked as answer]
Hi! Any reason you are not using a Redis Set (with [SADD](https://redis.io/commands/sadd) and [SREM](https://redis.io/commands/srem))?

**Reply by sajjad-fatehi** [reply]
thanks i used hset and hget with socket.id
